tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Rust.cs

```diff
@@ -76,10 +87,19 @@ public override async Task<IterationResult> RunAsync(
             string profilerOptions,
             object context)
         {
-            // Set AZURE_TEST_MODE environment variable to "live" before invoking the compiler
-            Environment.SetEnvironmentVariable("AZURE_TEST_MODE", "live");
-            // set up the params for cargo
-            string finalParams = $"test --release --package {primaryPackage} --test perf -- --test-results {_targetResultsDirectory}/{testName}-results.json \"{testName}\" {arguments}";
+            // The Rust SDK uses environment variables to determine if the tests should run in playback or live mode.
+            // If --test-proxy is in the command line arguments, set AZURE_TEST_MODE environment variable to "playback" to indicate that we are going to use the test proxy.
+            // Otherwise set test mode to `live`.
+            if (arguments.Contains("test-proxy"))
+            {
+                Environment.SetEnvironmentVariable("AZURE_TEST_MODE", "playback");
+            }
+            else
+            {
+                Environment.SetEnvironmentVariable("AZURE_TEST_MODE", "live");
+            }
```

Comment:
These cause a rebuild - at least for recorded tests. That's part of how the `#[recorded::test]` attribute works. Not sure if/how you'd change that. So this really needs to be during the build phase. But if you *only* need these at runtime, sure, this works.

Reply:
It rebuilds `azure_core_test` on up, so that's at least 3 crates. Per our offline discussion, I think we could probably avoid this, though.